testsuite/tests/templates/dunelike/gen-native.sh

```diff
@@ -9,4 +26,8 @@ s/\.cmo/.cmx/g
 s/\.bc/.exe/g
 # Do this last to avoid replacing the _byte:
 s/TEST/TEST (* DO NOT EDIT. Instead edit test_byte.ml and run gen-native.sh. *)/
-' test_byte.ml > test_native.ml
+' test_byte.ml > "$OUTPUT"
+
+if [ -n "$CHECKING" ]; then
+  diff -q test_native.ml "$OUTPUT" && rm "$OUTPUT"
```

Comment:
This should return non-zero to indicate error?

Reply:
It does; I've added a comment since it's not obvious.